src/java.desktop/share/classes/javax/swing/ImageIcon.java

```diff
@@ -367,10 +371,14 @@ public Image getImage() {
 
     /**
      * Sets the image displayed by this icon.
+     * Setting a {@code null} image will not render any image icon.
```

Comment:
```suggestion
     * Setting a {@code null} image renders no icon.
```
I think this is better: it's shorter, it uses the present tense because it's fact and it doesn't depend on anything.

Reply:
I think the existing text was OK. And "will" is better because it hasn't happened yet.
Rendering will happen later. Not when you construct this object.
My personally preferred wording would be something like
Setting a {@code null} image means any existing image will be removed and no image will be rendered.

Now .. what should happen to an existing description when you set a new image ?

Apps can call setDescription() but in what order ? 

The only thing I can say here is that I think if the image is null we probably should clear the description.

Reply:
I have modified the wording as per suggestion and also clearing the description now if image is null..